app/src/main/java/tech/relaycorp/letro/repository/PreferenceDataStoreRepository.kt

```diff
@@ -0,0 +1,48 @@
+package tech.relaycorp.letro.repository
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+@Singleton
+class PreferencesDataStoreRepository @Inject constructor(@ApplicationContext val context: Context) {
+
+    private val Context.dataStore by preferencesDataStore(name = "letro_preferences")
+
+    private val preferencesDataStore: DataStore<Preferences> = context.dataStore
+
+    private val firstPartyEndpointKey = stringPreferencesKey("firstPartyEndpointNodeId")
+    private val thirdPartyEndpointKey = stringPreferencesKey("thirdPartyEndpointNodeId")
```

Comment:
I think we should qualify these by prefixing them with `server`, and we could also omit `Node`:

```suggestion
    private val serverFirstPartyEndpointKey = stringPreferencesKey("firstPartyEndpointId")
    private val serverThirdPartyEndpointKey = stringPreferencesKey("thirdPartyEndpointId")
```